src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+    private final AnomalyDetector anomalyDetector;
+
+    private final Logger logger = LogManager.getLogger(IndexAnomalyDetectorActionHandler.class);
+    private final Integer maxAnomalyDetectors;
+    private final Integer maxAnomalyFeatures;
+    private final TimeValue requestTimeout;
+    private final NamedXContentRegistry xContent;
+
+    private ValidateResponse responseValidate;
+    private Map<String, List<String>> failuresMap;
+    private Map<String, List<String>> suggestedChangesMap;
+    private Boolean inferringInterval;
+    private AtomicBoolean inferAgain;
+
+    /**
+     * Constructor function.
+     *
+     * @param client                  ES node client that executes actions on the local node
+     * @param channel                 ES channel used to construct bytes / builder based outputs, and send responses
+     * @param anomalyDetectionIndices anomaly detector index manager
+     * @param anomalyDetector         anomaly detector instance
+     */
+    public ValidateAnomalyDetectorActionHandler(
+        NodeClient client,
+        RestChannel channel,
+        AnomalyDetectionIndices anomalyDetectionIndices,
+        AnomalyDetector anomalyDetector,
+        Integer maxAnomalyDetectors,
+        Integer maxAnomalyFeatures,
+        TimeValue requestTimeout,
+        NamedXContentRegistry xContentRegistry
+    ) {
+        super(client, channel);
+        this.anomalyDetectionIndices = anomalyDetectionIndices;
+        this.anomalyDetector = anomalyDetector;
+        this.maxAnomalyDetectors = maxAnomalyDetectors;
+        this.maxAnomalyFeatures = maxAnomalyFeatures;
+        this.requestTimeout = requestTimeout;
+        this.responseValidate = new ValidateResponse();
+        this.xContent = xContentRegistry;
+        this.inferringInterval = false;
+        this.inferAgain = new AtomicBoolean(true);
+        this.failuresMap = new HashMap<>();
+        this.suggestedChangesMap = new HashMap<>();
+    }
+
+    /**
+     * Start function to process validate anomaly detector request.
+     * Checks if anomaly detector index exist first, if not, add it as a failure case.
+     *
+     * @throws IOException IOException from {@link AnomalyDetectionIndices#initAnomalyDetectorIndexIfAbsent(ActionListener)}
+     */
+    public void startValidation() throws IOException {
+        if (!anomalyDetectionIndices.doesAnomalyDetectorIndexExist()) {
+            anomalyDetectionIndices
+                .initAnomalyDetectorIndex(
+                    ActionListener.wrap(response -> onCreateMappingsResponse(response), exception -> onFailure(exception))
+                );
+        } else {
+            preDataValidationSteps();
+        }
+    }
+
+    public void preDataValidationSteps() {
+        List<String> missingFields = new ArrayList<>();
+        List<String> formatErrors = new ArrayList<>();
+        if (anomalyDetector.getName() == null || anomalyDetector.getName() == "") {
```

Comment:
you may use `StringUtils.isBlank()` to check this

Reply:
fixed